etl/steps/data/garden/war/2023-06-22/ucdp.py

```diff
@@ -0,0 +1,300 @@
+    """
+    tb_conflict = tb_conflict[["conflict_id", "year", "type_of_conflict"]].drop_duplicates()
+    assert tb_conflict.groupby(["conflict_id", "year"]).size().max() == 1, "Some conflict_id-year pairs are duplicated!"
+
+    # Add `type_of_conflict` to `tb_geo`.
+    # This column contains the type of state-based conflict (1: inter-state, 2: intra-state, 3: extra-state, 4: internationalized intrastate)
+    tb_geo = tb_geo.merge(
+        tb_conflict, left_on=["conflict_new_id", "year"], right_on=["conflict_id", "year"], how="outer"
+    )
+    # Fill unknown types of violence
+    msk = tb_geo["type_of_violence"] == 1
+    tb_geo.loc[msk, "type_of_conflict"] = tb_geo.loc[msk, "type_of_conflict"].fillna("state-based (unknown)")
+
+    # Assert that `type_of_conflict` was only added for state-based events
+    assert tb_geo[tb_geo["type_of_violence"] != 1]["type_of_conflict"].isna().all()
+    # Check that `type_of_conflict` is not NaN for state-based events
+    assert (
+        not tb_geo[tb_geo["type_of_violence"] == 1]["type_of_conflict"].isna().any()
+    ), "Could not find the type of conflict for some state-based conflicts!"
+
+    # Create `conflict_type` column as a combination of `type_of_violence` and `type_of_conflict`.
+    type_of_violence_mapping = {
+        2: "non-state conflict",
+        3: "one-sided violence",
+    }
+    type_of_conflict_mapping = {
+        1: "extrasystemic",
+        2: "interstate",
+        3: "intrastate",
+        4: "internationalized intrastate",
+    }
+    tb_geo["conflict_type"] = (
+        tb_geo["type_of_conflict"]
+        .replace(type_of_conflict_mapping)
+        .fillna(tb_geo["type_of_violence"].replace(type_of_violence_mapping))
+    )
+
+    # Sanity check
+    assert tb_geo["conflict_type"].isna().sum() == 0, "Check NaNs in conflict_type (i.e. conflicts without a type)!"
+    return tb_geo
+
+
+def add_year_start(tb: Table) -> Table:
+    """Add year of conflict start."""
+    tb_start_year = tb.groupby("conflict_new_id", as_index=False)[["year"]].min()
+    tb = tb.merge(tb_start_year, on="conflict_new_id", how="left", suffixes=("", "_start"))
+    assert tb.year_start.isna().sum() == 0, "Check NaNs in year_start!"
+    return tb
+
+
+def add_number_conflicts_and_deaths(tb: Table) -> Table:
+    """Add number of ongoing and new conflicts."""
+    # Get number of ongoing conflicts, and deaths in ongoing conflicts
+    log.info("ucdp: get number of ongoing conflicts and deaths in ongoing conflicts")
+    tb_ongoing = _add_number_ongoing_conflicts_and_deaths(tb)
+
+    # Get number of new conflicts every year
+    log.info("ucdp: get number of new conflicts every year")
+    tb_new = _add_number_new_conflicts(tb)
+    # Combine and build single table
+    log.info("ucdp: combine and build single table")
+    tb = tb_ongoing.merge(
+        tb_new, left_on=["year", "region", "conflict_type"], right_on=["year_start", "region", "conflict_type"]
```

Comment:
Are the UCDP regions the same as ours? If not, it might be appropriate to signal this in some way or to recalculate the values for OWID continents?